locations/spiders/hannaford.py

```diff
@@ -1,122 +1,30 @@
-                            else:
-                                cleaned_times += ":".join(hour_min)
-                        else:
-                            if ampm[0]:
-                                cleaned_times += (":".join(hour_min)) + "-"
-                            else:
-                                cleaned_times += ":".join(hour_min)
-            else:
-                cleaned_times += "off"
-            hours[i] = cleaned_times
-        return hours
-
-    def parse_store(self, response):
-        storeid = response.url.split("=")[1]
-        for i in response.meta["lat_lon"]:
-            if i[0] == storeid:
-                lat = i[1]
-                lon = i[2]
-
-        addr = response.xpath('//div[@class="storeContact"]/p[1]/text()').extract()
-        addr = [x.strip().replace("\xa0", "-") for x in addr]
-        street = ""
-        cty_st_zip = ""
-        for i in addr:
-            search = re.search(regex, i)
-            if not search:
-                street += i + " "
-            else:
-                cty_st_zip += i
-
-        city = cty_st_zip.split(",")[0]
-        state = cty_st_zip.split("-")[0].split(",")[1]
-        postcode = cty_st_zip.split("-")[1]
-        addr_full = "{}{},{} {}".format(street, city, state, postcode)
-
-        phone = response.xpath('//div[@class="storeContact"]/p[2]/text()').extract_first().split("\xa0")[1]
-        name = response.xpath('//h1[@style="display:block;"]/text()').extract_first().split(" to ")[1]
-        website = response.xpath('//div[@class="storeContact"]/descendant::*/a/@href').extract_first()
-        if not website:
-            website = response.url
-        hours = response.xpath('//div[@class="storeHours"]/div/p/descendant::*/text()').extract()
-
-        opening_hours = self.combine_hours(hours)
-
-        yield GeojsonPointItem(
-            ref=response.url,
-            name=name,
-            lat=lat,
-            lon=lon,
-            addr_full=addr_full,
-            street=street,
-            city=city,
-            state=state,
-            postcode=postcode,
-            phone=phone,
-            website=website,
-            opening_hours=opening_hours,
-        )
+    allowed_domains = ["hannaford.com"]
+    sitemap_urls = ["https://stores.hannaford.com/sitemap.xml"]
+    sitemap_rules = [(r"[0-9]+$", "parse_sd")]
+    wanted_types = ["GroceryStore"]
+
+    def inspect_item(self, item, response):
+        days = json.loads(response.xpath('//h4[text()="Store Hours"]/following::div[1]/@data-days').get())
+        oh = OpeningHours()
+        for day in days:
+            opentime = str(day.get("intervals")[0].get("start")).zfill(4)
+            closetime = str(day.get("intervals")[0].get("end")).zfill(4)
+            oh.add_range(
+                day=day.get("day"),
+                open_time=f"{opentime[:2]}:{opentime[2:]}",
+                close_time=f"{closetime[:2]}:{closetime[2:]}",
```

Comment:
This could also be achieved with a custom `time_format`:
```suggestion
                open_time=opentime,
                close_time=closetime,
                time_format="%H%M"
```